src/proj6BittingCerratoCohenEllmer/controllers/CodeTabController.java

```diff
@@ -30,6 +35,50 @@
     private final HashMap<Tab,String> savedPaths = new HashMap<>();
 
     private final DialogHelper dialogHelper = new DialogHelper();
+    
+    // Vim Command Parameters
+    private boolean inVIMCommandMode = false;
+
+    //An Arraylist to keep track of commands for VIM chaining.
+    private final ArrayList<String> vimCommands = new ArrayList<String>();
+
+    private final EventHandler<KeyEvent> vimHandler = new EventHandler<KeyEvent>() {
+        /**
+         * Route VIM key press events to implemented methods
+         * 
+         * @param key A KeyEvent object that gives information about a Key Stroke
+         */
+        @Override
+        public void handle(KeyEvent key) {
```

Comment:
I feel like this makes the most sense in the subclass of the tab class as part of the instantiation process.